tests/components/humidifier/test_device_condition.py

```diff
@@ -0,0 +1,314 @@
+                        "data_template": {
+                            "some": "is_off {{ trigger.%s }}"
+                            % "}} - {{ trigger.".join(("platform", "event.event_type"))
+                        },
+                    },
+                },
+                {
+                    "trigger": {"platform": "event", "event_type": "test_event3"},
+                    "condition": [
+                        {
+                            "condition": "device",
+                            "domain": DOMAIN,
+                            "device_id": "",
+                            "entity_id": "humidifier.entity",
+                            "type": "is_mode",
+                            "mode": "away",
+                        }
+                    ],
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "some": "is_mode - {{ trigger.platform }} - {{ trigger.event.event_type }}"
+                        },
+                    },
+                },
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("humidifier.entity").state == STATE_ON
+    assert len(calls) == 0
+
+    hass.bus.async_fire("test_event1")
+    hass.bus.async_fire("test_event2")
+    await hass.async_block_till_done()
+    assert len(calls) == 1
+    assert calls[0].data["some"] == "is_on event - test_event1"
+
+    hass.states.async_set("humidifier.entity", STATE_OFF)
+    hass.bus.async_fire("test_event1")
+    hass.bus.async_fire("test_event2")
+    await hass.async_block_till_done()
+    assert len(calls) == 2
+    assert calls[1].data["some"] == "is_off event - test_event2"
+
+    hass.states.async_set(
+        "humidifier.entity", STATE_ON, {const.ATTR_MODE: const.MODE_AWAY}
+    )
+
+    hass.bus.async_fire("test_event3")
+    await hass.async_block_till_done()
+
+    assert len(calls) == 3
+    assert calls[2].data["some"] == "is_mode - event - test_event3"
+
+    hass.states.async_set(
+        "humidifier.entity", STATE_ON, {const.ATTR_MODE: const.MODE_HOME}
+    )
+
+    # Should not fire
+    hass.bus.async_fire("test_event3")
+    await hass.async_block_till_done()
+    assert len(calls) == 3
+
+
+async def test_capabilities(hass):
+    """Test capabilities."""
+    hass.states.async_set(
+        "humidifier.entity",
+        STATE_ON,
+        {
+            const.ATTR_MODE: const.MODE_AWAY,
+            const.ATTR_AVAILABLE_MODES: [const.MODE_HOME, const.MODE_AWAY],
+        },
+    )
+
+    # Test mode
+    capabilities = await device_condition.async_get_condition_capabilities(
+        hass,
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "device_id": "",
+            "entity_id": "humidifier.entity",
+            "type": "is_mode",
+        },
+    )
+
+    assert capabilities and "extra_fields" in capabilities
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)